cachi2/core/package_managers/yarn_classic/resolver.py

```diff
@@ -1,8 +1,25 @@
+import re
 from pathlib import Path
 from typing import Optional, Union
+from urllib.parse import urlparse
 
+from pyarn.lockfile import Lockfile
+from pyarn.lockfile import Package as PYarnPackage
 from pydantic import BaseModel
 
+from cachi2.core.errors import PackageRejected, UnexpectedFormat
+from cachi2.core.package_managers.npm import NPM_REGISTRY_CNAMES
+from cachi2.core.rooted_path import RootedPath
+
+GIT_HOSTS = frozenset(("github.com", "gitlab.com", "bitbucket.com", "bitbucket.org"))
```

Comment:
Added the link to the yarn source

> I wonder what Yarn does if it encounters a project from gitea

It should work, it would just need to match one of the GIT_PATTERN_MATCHERS :man_shrugging: